postprocessing/science/compute_isochrones.py

```diff
@@ -0,0 +1,244 @@
+    if "ASR" in sliprateParameters:
+        if "SRs" in sliprateParameters and "SRd" in sliprateParameters:
+            sliprates.append(np.sqrt(sliprates[0]**2 + sliprates[1]**2))
+        else:
+            sliprates.append(np.sqrt(faultXdmf.ReadData("SRs")[timeIndicesFault[0]:timeIndicesFault[1]].T**2 +
+                                     faultXdmf.ReadData("SRd")[timeIndicesFault[0]:timeIndicesFault[1]].T**2))
+    return np.array(sliprates)
+
+
+def ClosestReceiverIndex(coords, receiver_array):
+    dist = np.sum((receiver_array-coords)**2, axis=1)
+    return np.argmin(dist)
+
+
+def PTraveltime(receiverIndex, trigger=0.01):
+    """Picks the P-wave arrival. This function is only suited for synthetic data with negligible noise"""
+    
+    timeIndices = np.int_(timeIndicesFault * (dtFault / dtSurface))
+    absWaveform = np.sqrt(surfaceXdmf.ReadDataChunk(surfaceVariables[0], receiverIndex, 1)**2 +
+                      surfaceXdmf.ReadDataChunk(surfaceVariables[1], receiverIndex, 1)**2 +
+                      surfaceXdmf.ReadDataChunk(surfaceVariables[2], receiverIndex, 1)**2)[timeIndices[0]:timeIndices[1]]
+    
+    threshold = np.amax(absWaveform)*trigger           # trigger dictates the portion of the maximum ground velocity,
+    t = np.argmax(absWaveform>threshold)*dtSurface     # which needs to be reached to pick the p-phase
+    return t                                           # p-phase arrival time (equal to traveltime if nucleation at t=0)
+
+
+def ApproximateHypocenter(ruptureTimes, faultMidPoints):
+    
+    nucleationTime = dtFault
+    indHypocenter = (ruptureTimes > 0) & (ruptureTimes <= nucleationTime)
+    
+    # Take the mean of the first 100+ slipping elements to reduce impact from degenerated elements
+    while indHypocenter.sum() < 100:
+        nucleationTime += dtFault
+        indHypocenter = (ruptureTimes > 0) & (ruptureTimes <= nucleationTime)
+    
+    hypocenter = np.mean(faultMidPoints[indHypocenter,:], axis=0)
+    return hypocenter
+
+
+def GetTraveltimesToAllFaultCells(receiverCoords, faultMidPoints, swaveVelocity):
+    
+    distances = np.sqrt(np.sum((receiverCoords - faultMidPoints) ** 2, axis=1))
+    return distances * (1 / swaveVelocity)
+
+
+def MedianSmoothing(input_tuple):
+    # input_tuple: (parameter, xyz, indices, radius)
+    parameter = input_tuple[0]
+    xyz = input_tuple[1]
+    indices = input_tuple[2]
+    radius = input_tuple[3]
+    parameter_new = np.zeros(indices.shape[0])
+    for i, j in enumerate(indices):
+        parameter_new[i] = np.median(parameter[(xyz[:,0] < xyz[j,0]+radius) & (xyz[:,0] > xyz[j,0]-radius) & 
+                            (xyz[:,1] < xyz[j,1]+radius) & (xyz[:,1] > xyz[j,1]-radius) &                    
+                            (xyz[:,2] < xyz[j,2]+radius) & (xyz[:,2] > xyz[j,2]-radius)])
+    return parameter_new
+
+
+def MedianSmoothingParallel(parameter, xyz, radius=250., nprocs=4):
```

Comment:
would be clear if parameter was called
peakSliprateTimes (or a slightly different name if you want to avoid the name of the global value)